jujugui/static/gui/src/app/utils/environment-change-set.js

```diff
@@ -505,7 +506,16 @@ YUI.add('environment-change-set', function(Y) {
           services.getById(command.args[0]).set('deleted', false);
           break;
         case '_destroyMachines':
+          var machineId = command.args[0][0];
+          // Set the parent machine to not be deleted.
           machines.getById(command.args[0]).deleted = false;
+          // Set the containers of the parent machine to not be deleted.
+          machines.filterByAncestor(machineId).forEach(machine => {
+            machine.deleted = false;
+          });
+          units.filterByMachine(machineId, true).forEach(unit => {
+            unit.deleted = false;
```

Comment:
Is it OK to update the units/machines etc. here without reviving them? I guess the db events must get fired correctly otherwise this wouldn't work...